docs/freedesktop-quick-reference.rst

```diff
@@ -37,25 +37,53 @@ command, *Type* and *Icon*:
 
     [Desktop Entry]
     Name=Gnome Dictionary
-    Exec=org.gnome.Dictionary
+    Exec=gnome-dictionary
     Type=Application
     Icon=org.gnome.Dictionary
 
 Your desktop file should be prefixed with your application's appid and
-placed in ``/app/share/applications/``, you should use
-``desktop-file-validate`` to check your file for errors before including
-it.
+placed in ``/app/share/applications/`` within your app.
 
 Example:
 
 ::
 
     /app/share/applications/org.gnome.Dictionary.desktop
 
-You can find more information
-`here <https://wiki.archlinux.org/index.php/desktop_entries>`__. If
-interested, you can read also the full spec
-`here <https://standards.freedesktop.org/desktop-entry-spec/latest/>`__.
+It's recommended to use ``desktop-file-validate`` to check your file
+for errors before including it.
+
+A special note about the ``Exec`` line: When installing the app on a user's
+system, Flatpak will automatically transform your desktop file so that the app
+will be started through Flatpak. For example, when installing
+``org.gnome.Dictionary``, a file will be created at
+``/var/lib/flatpak/exports/share/applications/org.gnome.Dictionary.desktop``.
```

Comment:
Thanks for this PR! I'm in the same situation as you, couldn't figure out the `Exec` field of the desktop file based on the docs alone. Now my desktop file finally works!

One suggestion I would have, from the perspective of a reader, is I would remove "a file will be created at ...". To me, that makes it sound like there's some Flatpak magic specific to that file going on, creating that file for you. The previous sentence about "automatic transforms" seems to suggest that. Whereas it's actually you who put that file there by placing it in `/app/share/applications` inside your yaml, probably through a `install -D` or similar. On an installed system, the path corresponding to `/app/share/applications` is simply `/var/lib/flatpak/exports/share/applications`, but that's unrelated to the desktop file or the `Exec` line and the "automatic transformation", for example if you put something in `/app/foo/bar`, it would end up in `/var/lib/flatpak/exports/foo/bar`.

I would say something among the lines of:

> For example, when installing `org.gnome.Dictionary`, the `Exec` line is automatically transformed into this:
>
>      Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=gnome-dictionary org.gnome.Dictionary
>
> (If you placed the desktop file at `/app/share/applications/org.gnome.Dictionary.desktop` inside your Flatpak, the equivalent path on the installed system where you can view the transformed file would be  `/var/lib/flatpak/exports/share/applications/org.gnome.Dictionary.desktop`.)

Reply:
> if you put something in /app/foo/bar, it would end up in /var/lib/flatpak/exports/foo/bar.

This is incorrect. Flatpak only exports files at well-defined paths conditional on if they have allowed filenames. See for example: https://github.com/flatpak/flatpak/pull/4801

Reply:
OK, it was a bad example. But was my underlying point clear, that there's nothing magic about that file?

Reply:
I guess it depends what you mean by magic, but Flatpak does process the `.desktop` file specially; that's how it rewrites the `Exec` line. Perhaps we should phrase it like this:

"The .desktop file included in your app will be rewritten by Flatpak and exported to a path such as `exports/share/applications/org.gnome.Dictionary.desktop` under your Flatpak installation directory."

Reply:
I updated the wording based on the suggestion by @mwleeds.